src/Compilers/CSharp/Portable/Symbols/Symbol_Attributes.cs

```diff
@@ -517,6 +517,30 @@ static bool isObsoleteDiagnostic(DiagnosticWithInfo d)
                 return d.HasLazyInfo ? d.LazyInfo is LazyObsoleteDiagnosticInfo : d.Info.IsObsoleteDiagnostic();
             }
         }
+
+        /// <summary>
+        /// Binds attributes applied to this symbol.
+        /// </summary>
+        protected ImmutableArray<(CSharpAttributeData, BoundAttribute)> BindAttributes(OneOrMany<SyntaxList<AttributeListSyntax>> attributeDeclarations, Binder rootBinder)
+        {
+            var binder = new ContextualAttributeBinder(rootBinder, this);
+            var boundAttributeArrayBuilder = ArrayBuilder<(CSharpAttributeData, BoundAttribute)>.GetInstance();
+            foreach (var attributeListSyntaxList in attributeDeclarations)
+            {
+                foreach (var attributeListSyntax in attributeListSyntaxList)
+                {
+                    foreach (var attributeSyntax in attributeListSyntax.Attributes)
+                    {
+                        var boundType = binder.BindType(attributeSyntax.Name, BindingDiagnosticBag.Discarded);
+                        var boundTypeSymbol = (NamedTypeSymbol)boundType.Type;
+                        var boundAttribute = binder.GetAttribute(attributeSyntax, boundTypeSymbol,
```

Comment:
>binder [](http://example.com/codeflow?start=45&length=6)

The previous logic (from `SourceComplexParameterSymbol.BindParameterAttributes`) had 
```
var boundAttribute = new ExecutableCodeBinder(attributeSyntax, binder.ContainingMemberOrLambda, binder)
                            .BindAttribute(attributeSyntax, boundTypeSymbol, this, BindingDiagnosticBag.Discarded);
```

What's the impact/intent of this change?

Reply:
Good catch, I forgot to explain. `binder.GetAttribute(...)` internally calls `new ExecutableCodeBinder(attributeSyntax, binder.ContainingMemberOrLambda, binder).BindAttribute(...)` which I noticed here and simplified the code.

Reply:
Thanks for the clarification.